.github/workflows/fourmolu.yaml

```diff
@@ -0,0 +1,52 @@
+# Workflow triggered by pushes to the main branch and PRs targeting it
```

Comment:
Can we specify that the build job should wait for this to complete?

The build job is very resource intensive and we should ideally not run it if formatting fails.

Reply:
I did consider that.
It would mean that I have to put the formatting as a job into the same workflow file as the build.
If you don't mind I would prefer to do that in another PR, so the formatting of the code can be merged.